File: frontend/src/app/services/exchange/exchange.service.ts
import {HttpClient} from '@angular/common/http';
import {Injectable} from '@angular/core';
import {Observable} from 'rxjs';
import {MessageResponse} from '../../model/messageResponse';
import {StockExchange} from '../../model/StockExchange';
import {CreateExchange} from "../../model/create-exchange";

const API_URL = 'http://localhost:8080/exchange/';

@Injectable({
  providedIn: 'root'
})
export class ExchangeService {

  constructor(private http: HttpClient) {
  }

  getExchange(page: number, size: number, sort: string[], ownerId: number): Observable<any> {
    return this.http.get(API_URL + 'find-by-owner' + `?page=${page}&size=${size}&sort=${sort}&ownerId=${ownerId}`);
  }

  get(id: number): Observable<any> {
    return this.http.get(API_URL + 'find/' + `${id}`);
  }

  changeStatus(id: number, status: string): Observable<MessageResponse> {
    return this.http.post<MessageResponse>(API_URL + 'change-status' + `?id=${id}&status=${status}`, null);
  }

  updateExchange(stockExchange: StockExchange): Observable<MessageResponse> {
    console.log(stockExchange);
    return this.http.post<MessageResponse>('http://localhost:8080/exchange/update', stockExchange);
  }

  saveExchange(exchange: CreateExchange, id: number): Observable<MessageResponse> {
    console.log(exchange);
    return this.http.post<MessageResponse>(API_URL + 'create' + '?owner-id=' + id, exchange);
  }

  getAll(): Observable<any> {
    return this.http.get(API_URL + 'find/all');
  }

  getStatistics(page: number, size: number, exchangeId: number): Observable<any> {
    return this.http.get(API_URL + 'find/statistics' + `?page=${page}&size=${size}&exchangeId=${exchangeId}`);
  }
}

File: fapi/src/main/java/stock_exchange/service/ExchangeService.java
package stock_exchange.service;

import stock_exchange.model.CreateStockExchange;
import stock_exchange.model.OwnerStatistics;
import stock_exchange.model.StockExchange;
import stock_exchange.model.response.MessageResponse;
import stock_exchange.model.response.PageResponse;

import java.util.List;

public interface ExchangeService {
    PageResponse<StockExchange> findAll(int page, int size, String[] sort, int ownerId);

    List<StockExchange> findAll();

    StockExchange find(int exchangeId);

    MessageResponse changeStatus(int exchangeId, String status);

    MessageResponse update(StockExchange exchange);

    MessageResponse create(int ownerId, CreateStockExchange exchange);

    PageResponse<OwnerStatistics> getStatistics(int exchangeId, int page, int size);
}
